python/ql/src/meta/alerts/RemoteFlowSourcesReach.ql

```diff
@@ -0,0 +1,46 @@
+/**
+ * @name Remote flow sources reach
+ * @description Nodes that can be reached with taint tracking from sources of
+ *   remote user input.
+ * @kind problem
+ * @problem.severity recommendation
+ * @id py/meta/alerts/remote-flow-sources-reach
+ * @tags meta
+ * @precision very-low
+ */
+
+private import python
+private import semmle.python.dataflow.new.DataFlow
+private import semmle.python.dataflow.new.TaintTracking
+private import semmle.python.dataflow.new.RemoteFlowSources
+private import meta.MetaMetrics
+
+class RemoteFlowSourceReach extends TaintTracking::Configuration {
+  RemoteFlowSourceReach() { this = "RemoteFlowSourceReach" }
+
+  override predicate isSource(DataFlow::Node node) {
+    node instanceof RemoteFlowSource and
+    not node.getLocation().getFile() instanceof IgnoredFile
+  }
+
+  override predicate isSink(DataFlow::Node node) {
+    not node.getLocation().getFile() instanceof IgnoredFile and
+    (
+      node instanceof RemoteFlowSource
+      or
+      this.isAdditionalFlowStep(_, node)
+    ) and
+    // we used to do `obj -> obj.meth` and `obj.meth -> obj.meth()` in two separate
+    // steps, and now do them in one `obj -> obj.meth()`. So we're going to ignore the
+    // fact that we no longer taint the node in the middle.
+    not exists(DataFlow::MethodCallNode c |
+      node = c.getFunction() and
+      this.isAdditionalFlowStep(c.getObject(), node) and
+      this.isAdditionalFlowStep(node, c)
+    )
```

Comment:
Since I'm using meta-queries in the DCA run [internal link](https://github.com/dsp-testing/RasmusWL-dca/blob/run/ce4b192caaf29f663fb77cd7063d5f310ea03946__Differences__code-scanning/reports/alert-meta-comparison.md), and I was comparing what used to be reachable from remote-flow-sources to what we do now (to see how much the modeling effort improved things), then for every case like `meth.foo()` we would no longer taint `meth.foo`, so that would be counted as _one less thing reachable_, which completely offset the comparison. So that's why we are excluding `meth.foo`

Reply:
Aha, so this is new code affecting old behaviour, making it match the new one. That is why the comment feels opposite to the code.

Reply:
Could we make a separate PR for just removing that bit (changing nothing else) to avoid keeping this piece of code around? (Or is the thought perhaps to try to track these results over time?)

Reply:
If we want to track changes such as "how good have our support gotten compared to one year ago" it would be nice to keep this code around. Since this is only part of our internal meta queries, I don't think it's too bad keeping it around for a while.

I'm 

```suggestion
    // September 2021:
    // we used to do `obj -> obj.meth` and `obj.meth -> obj.meth()` in two separate
    // steps, and now do them in one `obj -> obj.meth()`. So we're going to ignore the
    // fact that we no longer taint the node in the middle.
    // We should be able to remove this code once we don't care to compare with this
    // (before September 2021) way of doing taint-propagation
    not exists(DataFlow::MethodCallNode c |
      node = c.getFunction() and
      this.isAdditionalFlowStep(c.getObject(), node) and
      this.isAdditionalFlowStep(node, c)
    )
```

Reply:
I made an alternative suggestion inspired by this one.

Reply:
I'm not 100% on board with that one, since we didn't change the underlying taint tracking library :flushed: 

How about this one?

```suggestion
    // In september 2021 we changed how we do taint-propagation for method calls (mostly
    // relating to modeled frameworks/libraries). We used to do `obj -> obj.meth` and
    // `obj.meth -> obj.meth()` in two separate steps, and now do them in one
    // `obj -> obj.meth()`. To be able to compare the overall reach between these two
    // version, we don't want this query to alert us to the fact that we no longer taint
    // the node in the middle (since that is just noise).
    // see https://github.com/github/codeql/pull/6349
    //
    // We should be able to remove the following few lines of code once we don't care to
    // compare with the old (before September 2021) way of doing taint-propagation for
    // method calls.
    not exists(DataFlow::MethodCallNode c |
      node = c.getFunction() and
      this.isAdditionalFlowStep(c.getObject(), node) and
      this.isAdditionalFlowStep(node, c)
    )
```

Reply:
I am happy with that :-)